File: elixir/advent2022/test/day01_test.exs
defmodule Advent2022.Day01.Test do
  import Advent2022.Day01
  use ExUnit.Case

  @sample_input "1000
2000
3000

4000

5000
6000

7000
8000
9000

10000"

  test "sample input" do
    burdens = @sample_input |> burdens_by_heaviest()
    assert sum_topN(burdens, 1) == 24000
    assert sum_topN(burdens, 3) == 45000
  end
end

File: kotlin/advent2022/app/src/test/kotlin/advent2022/Day01Test.kt
package advent2022

import kotlin.test.Test
import kotlin.test.assertNotNull
import kotlin.test.assertEquals

class Day01Test {
  val sampleInput = """
    1000
    2000
    3000

    4000

    5000
    6000

    7000
    8000
    9000

    10000""".trimIndent()

  @Test fun topBurden() {
    var burdens = Day01.readAndSortBurdens(sampleInput)
    assertEquals(24000, burdens.take(1).sum())
    assertEquals(45000, burdens.take(3).sum())
  }
}
